### PR TITLE
[Composer] Manage SF 2.8 compatibility (which remove symfony/propel1-bridge)

### PR DESCRIPTION
**Pull request by heristop**

### CODE REVIEW

File: Resources/doc/reference/model_filter.rst
Comment (greg0ire) [inline Resources/doc/reference/model_filter.rst]: Why alias this?

File: Admin/Admin.php
Comment (greg0ire) [inline Admin/Admin.php]: Well now there is no name collision, maybe just remove the alias?

File: Guesser/AbstractTypeGuesser.php
Comment (greg0ire) [inline Guesser/AbstractTypeGuesser.php]: I don't know anything about propel. Can you explain what is the difference between the bridge and the bundle and show why moving to the bundle is the way to go?

File: Tests/Fixtures/App/src/Sonata/TestBundle/Admin/BlogPostAdmin.php
Comment (greg0ire) [inline Tests/Fixtures/App/src/Sonata/TestBundle/Admin/BlogPostAdmin.php]: Same here, why alias this?

File: Tests/Functionnal/DashboardTest.php
Comment (greg0ire) [inline Tests/Functionnal/DashboardTest.php]: Why do you (have to) change this?

File: Guesser/AbstractTypeGuesser.php
Comment (heristop) [inline Guesser/AbstractTypeGuesser.php]: If you know nothing about Propel why are you a contributor on this porject? I use this bundle in production since 2 years.

Comment (greg0ire) [inline Guesser/AbstractTypeGuesser.php]: Because I'm a Sonata member, which grants me write access to all Sonata projects. It does not mean that I have to learn Propel before reviewing this though. Maybe another Sonata member can vouch for this kind of modification, but in the meantime, I thought you would be glad to get a thorough review of your changes. Was I wrong to think that?

File: composer.json
Comment (greg0ire) [inline composer.json]: Can you explain why you remove this particular dependency?

Comment (greg0ire) [inline composer.json]: Answering my own question : b/c it's a dependency of the admin bundle

File: composer.json
Comment (greg0ire) [inline composer.json]: Please use `^` here too, doing otherwise is a bit confusing.

File: composer.json
Comment (greg0ire) [inline composer.json]: Please sort dependencies alphabetically

File: composer.json
Comment (greg0ire) [inline composer.json]: Why do you remove the `provide` section?

File: Guesser/AbstractTypeGuesser.php
Comment (heristop) [inline Guesser/AbstractTypeGuesser.php]: This PR remove prope1-bridge which it uses on master with PropelTypeGuesser. It means this bundle cannot work with SF project >= 2.7. It means also we should drop SF obsolete versions.

Comment (greg0ire) [inline Guesser/AbstractTypeGuesser.php]: Ok so looking at [this issue](https://github.com/symfony/symfony/issues/13091), I can understand why.

> It means also we should drop SF obsolete versions.

That's the plan for every next major release of the sonata bundle (I think the requirement might even be sf 2.8).

Now regarding this issue we had previously, namely [your tweet](https://twitter.com/heristop/status/771746480742891521), and your previous [comment](https://github.com/sonata-project/SonataPropelAdminBundle/pull/71#discussion_r77372796), I think I understand why you reacting like this. You are taking some time to help us out with this bundle, and on behalf of all the Sonata team, I thank you for it, because I know what how time-consuming it can be to contribute something. That is maybe your way of giving back to this piece of open-source software, to show your gratefulness for the time it might have saved you.

With that in mind, I can understand how my many comments can be overwhelming, and may make me sound very distrustful and ungrateful to you. When merging, we do not do a background check on you to evaluate if you know how to code. Instead, we just try to read the code and ask questions until we fully understand it. We also try to make sure that **anyone** that reads your PR or better, your commits, can understand *why* you did every change. The thing is, when we click the merge button, the code you wrote becomes ours to maintain.

Now when we do not do that, when we merge blindly because "it looks good", [here is what happens](https://github.com/sonata-project/SonataMediaBundle/pull/909#discussion_r77162504). As you can see in this discussion, we get [issues](https://github.com/sonata-project/SonataMediaBundle/issues/1065) that are very hard to understand, and when we finally find the guy that introduced the faulty piece of code, what do we hear? ["I don't have time to look at this now. Since it's FOSS software, feel free to contribute. ;)"](https://github.com/sonata-project/SonataMediaBundle/issues/1065). So basically, this is our problem now. This guy made his PR to get his particular application to work, and when we ask for help, we consider ourselves lucky that he even answers to us (not everybody does that).

I guess you must be asking yourself some questions now. "How is it possible that no one knows how this work ?". Well that's wrong, there are people like @rande who do know how most of all this works. However, they are busy with their lives, or trying to keep up with the fuckton of other comments they have to review. Your contribution is maybe very nice, but if you get a review from him, you will be very lucky, and maybe your contribution will get merged fast because he fully understands it.

To spare you the frustration of feeling ignored, I come and try to understand your contribution. But since I do not want to be held accountable for bugs or problems I might have created by merging, I have no choice but to ask you some questions, that will help me understand both the PR and the software. I also am a grammar-nazi, very nitpicky about the commits and every single semicolon in your PR. I'll ask for explanation, tests and docs. While this might not be very useful when dealing with a small application, I deeply believe it is important for an open-source project. This makes of course the PR harder to merge **but** the comments I make, you will get them quickly. My first comment on this PR was literally 3 minutes after you made your PR. Do you know many project where this happens? Also, I try to make my comments *actionable*. Remove an alias? Easy-peasy! Changing a tilde to a caret? Piece of cake. Explaining what you did clearly? Shouldn't be a problem, and should be done in your commit messages (yes, I am using the plural here, not everything needs to be in one huge commit that says "fix bug").

So with all that in mind, I hope you are not discouraged, and I renew my commitment into giving you fast feedback, and all the help I can provide you with making this a no-brainer to merge. And you are overwhelmed and do not want to finish the job, I might even go as far as reworking your commits and doing it for you.

Comment (heristop) [inline Guesser/AbstractTypeGuesser.php]: I didn't expect this PR will be accepted blindly after the first submit (the travis config has not been updated yet), and it is ok for the review.

I just got the feeling that our project is the only one which use this bundle since the BC break of february. 
And the last sucessful build is 2 years old: https://travis-ci.org/sonata-project/SonataPropelAdminBundle/builds. So, I'm still thinking the risk is limited, considering this bundle is not stable. 

Anyway, I agree to have a sf 2.8 requirement and still have to push the new Travis config (to have a successful build for instance ;)).

Comment (greg0ire) [inline Guesser/AbstractTypeGuesser.php]: > I just got the feeling that our project is the only one which use this bundle since the BC break of february. 

Its not, [there are even packages that publicly depend on it, with last activity 6 month ago](https://packagist.org/packages/sonata-project/propel-admin-bundle/dependents)

But indeed [the stats are very very low](https://packagist.org/packages/sonata-project/propel-admin-bundle/stats) so let's not be too strict.

Good luck with the build!

**Comment by OskarStark** [comment]
@Soullivaneuh can we at least setup StyleCI for this project and remove the Scruitinizer?

**Comment by greg0ire** [comment]
Regarding the propel1 bridge, [I made them clarify the situation](https://github.com/symfony/symfony/issues/19833)

**Comment by core23** [comment]
Any news here?

**Comment by OskarStark** [comment]
@greg0ire any news here?

**Comment by greg0ire** [comment]
Uuuuuh… not from me. But it seems @heristop fixed the build.

**Comment by OskarStark** [comment]
> Uuuuuh… not from me. But it seems @heristop fixed the build.

so shall we merge this?  😕  ❓

**Comment by greg0ire** [comment]
Given the current state of this bundle, I'd say yes…

**Review by core23 — CHANGES_REQUESTED** [review]
Please update the PR template.

**Pull request closed without merging**
